src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -0,0 +1,445 @@
+            Assert.True(compilation3.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            m1 = compilation3.GetMember<MethodSymbol>("I1.M1");
+            test2 = compilation3.GetTypeByMetadataName("Test2");
+
+            Assert.Same(m1, test2.FindImplementationForInterfaceMember(m1));
+
+            compilation3.VerifyDiagnostics();
+            CompileAndVerify(compilation3, verify: false);
+        }
+
+        [Fact]
+        public void MethodImplementation_02()
+        {
+            var source1 =
+@"
+interface I1
+{
+    void M1() {}
+    void M2() {}
+}
+
+class Base
+{
+    void M1() { }
+}
+
+class Derived : Base, I1
+{
+    void M2() { }
+}
+
+class Test : I1 {}
+";
+            var compilation1 = CreateCompilationWithMscorlib(source1, options: TestOptions.DebugDll,
+                                                             parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation1.VerifyDiagnostics(
+                // (13,23): error CS0737: 'Derived' does not implement interface member 'I1.M2()'. 'Derived.M2()' cannot implement an interface member because it is not public.
+                // class Derived : Base, I1
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberNotPublic, "I1").WithArguments("Derived", "I1.M2()", "Derived.M2()").WithLocation(13, 23),
+                // (13,23): error CS0737: 'Derived' does not implement interface member 'I1.M1()'. 'Base.M1()' cannot implement an interface member because it is not public.
+                // class Derived : Base, I1
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberNotPublic, "I1").WithArguments("Derived", "I1.M1()", "Base.M1()").WithLocation(13, 23)
+                );
+        }
+
+        [Fact]
+        public void MethodImplementation_03()
+        {
+            var source1 =
+@"
+interface I1
+{
+    void M1() {}
+}
+
+class Test1 : I1
+{
+    public static void M1() { }
+}
+
+class Test2 : I1 {}
+";
+            var compilation1 = CreateCompilationWithMscorlib(source1, options: TestOptions.DebugDll,
+                                                             parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation1.VerifyDiagnostics(
+                // (7,15): error CS0736: 'Test1' does not implement interface member 'I1.M1()'. 'Test1.M1()' cannot implement an interface member because it is static.
+                // class Test1 : I1
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberStatic, "I1").WithArguments("Test1", "I1.M1()", "Test1.M1()").WithLocation(7, 15)
+                );
```

Comment:
The diagnostic is misleading at best, and possibly incorrect. The second sentence is true, but the first is not. It isn't clear that an error is required under these circumstances (though a warning might be helpful).